pkg/spanz/span.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package spanz
+
+import (
+	"bytes"
+
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb/pkg/ddl"
+	"github.com/pingcap/tidb/pkg/kv"
+	"github.com/pingcap/tidb/pkg/tablecodec"
+	"github.com/pingcap/tidb/pkg/util/codec"
+	"github.com/pingcap/tiflow/cdc/processor/tablepb"
+	"github.com/pingcap/tiflow/pkg/errors"
+	"go.uber.org/zap"
+)
+
+const (
+	// JobTableID is the id of `tidb_ddl_job`.
+	JobTableID = ddl.JobTableID
+	// JobHistoryID is the id of `tidb_ddl_history`
+	JobHistoryID = ddl.HistoryTableID
+)
+
+// UpperBoundKey represents the maximum value.
+var UpperBoundKey = []byte{255, 255, 255, 255, 255}
+
+// HackSpan will set End as UpperBoundKey if End is Nil.
+func HackSpan(span tablepb.Span) tablepb.Span {
+	if span.StartKey == nil {
+		span.StartKey = []byte{}
+	}
+
+	if span.EndKey == nil {
+		span.EndKey = UpperBoundKey
+	}
+	return span
+}
+
+// GetTableRange returns the span to watch for the specified table
+// Note that returned keys are not in memcomparable format.
+func GetTableRange(tableID int64) (startKey, endKey []byte) {
+	tablePrefix := tablecodec.GenTablePrefix(tableID)
+	sep := byte('_')
+	recordMarker := byte('r')
+
+	var start, end kv.Key
+	// ignore index keys.
+	start = append(tablePrefix, sep, recordMarker)
+	end = append(tablePrefix, sep, recordMarker+1)
+	return start, end
+}
+
+// GetAllDDLSpan return all cdc interested spans for DDL.
+func GetAllDDLSpan() []tablepb.Span {
```

Comment:
This can be deleted.

Reply:
The implementation related to tidb_ddl_history should not be placed in the spanz module, so we refactored a function, and here, the GetAllDDLSpan will be removed.